# **HOMEWORK 3**
## **DUE: *October 22, 2026 @ 11:59 PM***
## **24-HR LATE DUE DATE W/ A 15% PENALTY: *October 23, 2026 @ 11:59 PM***

### **DO NOT REMOVE ANY PART OF ANY OF THE QUESTIONS OR YOU LOSE CREDIT**
### *No Hardcoding Any Part of Any Question*
### **REMEMBER TO SHOW ALL CODE OUTPUT (NO CREDIT OTHERWISE)**
### **Reminder: Use tools or websites such as https://www.convert.ploomber.io/ to create a clean PDF from jupyter, check piazza for more details**
### **Reminder: Please make sure your code runs before submitting your work. Code sections that do not run will receive 0 credits, no partials will be given. This is VERY important in real project development.**

## **Part 1: Data Exploration (120 Points Total)**

---
---
**DISCLAIMER: This is a fake story, not a real incident! We have created this story just for the data exploration purpose.**

---
---

**Campus Police Department**

Incident Report

---

**Incident Date:** Jan 27, 2026

**Location of Incident:** 2nd Floor, Iribe Building

**Items Left at the Scene:** Red Ticket for Zichao's Magic Show

**Case Number:** 2027-IRB-0027

---
**Victim Information:**

**Name:** Dr. Fardina Alam

**Affiliation:** University of Maryland, College Park

**Position:** Professor

**Contact Information:** fardina@umd.edu

---

**Incident Description:**

On the evening of Jan 27, 2026, an unknown individual(s) unlawfully entered Dr. Fardina's room located on the 2nd floor of Iribe and stole a set of Markers. The stolen items were a set of limited edition markers that Dr. Fardina had spent years collecting. At the scene of the crime, a red entry ticket for "Zichao's Magic Show" was found on the floor near Dr. Fardina's desk.






In [4]:
# We import all essential libraries for this HW.
# If you want to add additional libraries, you can add more here!
import pandas as pd
from datetime import datetime
import string

## YOUR TASK:

In this assignment, you are tasked with solving the mystery of Dr. Fardina's stolen markers and finding the culprit.

* To accomplish this, you will be provided with **4 data sets** containing key information related to the incident. The list of the data sets name given below:

 - Profiles
 - Witnesses
 - Zichao's magic show
 - Interrogation Statements



* Your goal is to analyze these data sets to uncover the clues and evidences that will lead you to the thief.

**Be careful in following instructions as it may lead to you losing points :(**

Take a moment to look at all the data sets below:

---

**Profiles:**

This dataset contains information regarding all the potential people that could have committed the crime. General information regarding each person are as follows:
- `ID`: A **unique** identification for each person
- `Name`: The name of the person
- `Birthday`: The birthday of the person
- `Location_during_crime`: The location of the person during the crime
- `Eye_Color`: The eye color of the person
- `Skin_Color`: The skin color of the person
- `Hair_Color`: The hair color of the person
- `Height`: The height of the person in cm

**Witnesses:**

This dataset contains a statement from everyone in profiles, regarding suspicious things they saw during the time of the crime. Information that comes in this dataset are as follows:
- `ID`: A **Unique** identification for each person
- `Statement`: A statement from the person regarding a suspicious activity during the time of the crime

**Zichao's Magic Show:**

This dataset contains everyone that purchased a ticket for the show. **NOTE** that not everyone in profiles has purchased a ticket for this show. The Information contained in this dataset are as follows:
- `ID`: A **Unique** indentification for each person
- `Ticket_Number`: A **Unique** number that serves as an identification for the ticket bought

**Interrogation Statements:**

This datasets contains a statement from each person after they have been interrogated. **NOTE** assume that each statement is ***100% true*** and that ***there are people in this dataset who aren't in the "Profiles" dataset***.
Information contained in this dataset are as follows:

- `Name`: The name of the person being interrogated
- `Statement`: A statement the person gives after being interrogated

---

## Q1) Lets begin by observing the profiles dataset (10 PTS Total)


**TASK 1.1 (1 Point):**  In the code box below, load in `Profiles` dataset into a variable called `profiles_df` and display only top 5 rows.

**Outputs to be Graded:**
- Variable(s): `profiles_df`
- Cell outputs





In [5]:
# Write your code here
profiles_df = pd.read_csv('Profiles.csv')
profiles_df.head()

,ID,Name,Birthday,Location_during_crime,Eye_Color,Skin_Color,Hair_Color,Height
0,P299,Marco Rossi,1969/07/10,fourth floor,red,indigo,orange,191
1,P086,Aarav Patel,1979/03/10,second floor,indigo,indigo,green,198
2,P022,Li Wei,1990/10/08,fourth floor,blue,red,violet,154
3,P003,Haruto Tanaka,1996/27/10,lobby,green,blue,violet,158
4,P036,Kwame Mensah,1991/26/02,fourth floor,blue,yellow,orange,184


**TASK 1.2 (1 Point):**  Next, you need to display the number of rows and columns in the `profiles_df` dataframe.

**Outputs to be Graded:**
- Cell outputs

In [6]:
# Write your code here
profiles_df.shape

(348, 8)

**TASK 1.3 (1 Point):**  Next, you need to display the name of the columns in the `profiles_df` dataframe.

**Outputs to be Graded:**
- Cell outputs

In [7]:
# Write your code here
profiles_df.columns

Index(['ID', 'Name', 'Birthday', 'Location_during_crime', 'Eye_Color',
       'Skin_Color', 'Hair_Color', 'Height'],
      dtype='str')

**TASK 1.4 (1 Point):**  Now, you need to display the types of the columns in the `profiles_df` dataframe.

**Outputs to be Graded**
- Cell outputs

In [8]:
# Write your code here

profiles_df.dtypes

ID                         str
Name                       str
Birthday                   str
Location_during_crime      str
Eye_Color                  str
Skin_Color                 str
Hair_Color                 str
Height                   int64
dtype: object

**TASK 1.5 (2 Points):**
- Utilizing the `count()` function on this dataset and set it to the variable `info` and then display it.
- Write any **potential problem** that you see with the dataset when observing the output of the count function:

**Outputs to be Graded**
- Variable(s): `info`
- Cell outputs
- Written statement(s) describing potential problem(s)





In [9]:
# Write your code here

info = profiles_df.count()

info

ID                       348
Name                     348
Birthday                 348
Location_during_crime    280
Eye_Color                280
Skin_Color               348
Hair_Color               348
Height                   348
dtype: int64

**<font color='red'>Write your observations here:</font>** **Observations:** The `count()` output shows that `Location_during_crime` and `Eye_Color` each have only 280 non-null values out of 348 rows, so 68 values are missing in each column. This is a problem because missing locations make it harder to place suspects near the crime scene, and missing eye colors can cause us to exclude or misidentify potential culprits. VERY SUS!


**TASK 1.6 (4 Points):**

Look back at the "Profiles" dataset. Using other pandas functions than `count()`, **can you spot another potential problem with the dataset?**

**Outputs to be Graded**
- Written statement(s) describing your observations

In [10]:
# Write your code here 

print(f' Number of unique ids: {profiles_df["ID"].nunique()}')
print('========================================================================')
print(profiles_df[profiles_df['ID'] == 'P299'])

 Number of unique ids: 300
       ID             Name    Birthday Location_during_crime Eye_Color  \
0    P299      Marco Rossi  1969/07/10          fourth floor       red   
141  P299  Anushk Pokharna  1969/07/10          fourth floor       red   

    Skin_Color Hair_Color  Height  
0       indigo     orange     191  
141     indigo     orange     191  


**<font color='red'>Write your observations here:</font>** There are only 300 unique IDs across 348 rows, so 48 rows share an ID with another row. For example, ID P299 appears twice with different names (Marco Rossi and Anushk Pokharna) but identical birthday, location, eye, skin, and hair color, and height. So ID is not a reliable unique identifier.

## Q2) Cleaning the data set (25 PTS Total)

**TASK 2.1 (10 Points):** Lets begin by fixing the Birthday column. You may notice that the dates are in inconsistent format!

- Pandas internally represent dates in 'datetime64' format which is a form of ISO 8601. Basically, dates are represented as yyyy-mm-dd.


- Fix all the dates in the Birthday Column of `profiles_df` dataframe (Do not use other variable name).
- Display the `profiles_df` dataframe (only top 5 rows) after you are done.
- Note that the only **2 formats** the dates are in is either **yyyy/dd/mm** or **mm/dd/yyyy**;
- You want to represent them as **yyyy-mm-dd**

- **EXAMPLE**: In the “Profiles” dataset, dates like *2000/10/12* and *10/12/2000* follow different formats—one as *December 10th, 2000*, and the other as *October 12th, 2000*. We aim to standardize them to *2000-12-10* and *2000-10-12*, respectively.

**HINT:** some datetime functions might make things easier.
- https://www.geeksforgeeks.org/python-pandas-to_datetime/
- https://www.programiz.com/python-programming/datetime/strftime

**Outputs to be Graded**
- Variable(s): `profiles_df`
- Cell outputs


In [16]:
# Write your code 
#Chanign th date format to y/m/d

def format_func(date):
    s = ""
    count = 0;
    while  True:
        if date[count] == '/' and count > 2:
            return pd.to_datetime(date, format = '%Y/%d/%m')
        elif date[count] == '/' and (count == 2 or count == 1):
            return pd.to_datetime(date, format = '%m/%d/%Y')
        else:
            count += 1
            

series = profiles_df['Birthday'].apply(format_func)
profiles_df['Birthday'] = series

profiles_df.head()

,ID,Name,Birthday,Location_during_crime,Eye_Color,Skin_Color,Hair_Color,Height
0,P299,Marco Rossi,1969-10-07,fourth floor,red,indigo,orange,191
1,P086,Aarav Patel,1979-10-03,second floor,indigo,indigo,green,198
2,P022,Li Wei,1990-08-10,fourth floor,blue,red,violet,154
3,P003,Haruto Tanaka,1996-10-27,lobby,green,blue,violet,158
4,P036,Kwame Mensah,1991-02-26,fourth floor,blue,yellow,orange,184


**TASK 2.2 (5 Points):** Now that we have cleaned up the Birthday column, lets find the age of each person. Create a new column called `Age` that displays the age of the person **on the day of the crime.** Once you are done, display the
`profiles_df` dataframe  (only top 5 rows)

**Outputs to be Graded**
- Variable(s): `profiles_df`
- Cell outputs

In [18]:
# Write your code here

#turn Jan 27, 2026 into dateetime
day_of_crime = "2026-01-27"
day_of_crime = pd.to_datetime(day_of_crime) #turn the day of the crime into a datetime obj

profiles_df["Age"] = day_of_crime  - profiles_df['Birthday'] #return time delta whos format is  days, then hours:minutes:seconds

profiles_df["Age"] = (profiles_df["Age"].dt.days // 365.25).astype(int) #get just the days and divide by 365.25 to get the age

profiles_df.head()

,ID,Name,Birthday,Location_during_crime,Eye_Color,Skin_Color,Hair_Color,Height,Age
0,P299,Marco Rossi,1969-10-07,fourth floor,red,indigo,orange,191,56
1,P086,Aarav Patel,1979-10-03,second floor,indigo,indigo,green,198,46
2,P022,Li Wei,1990-08-10,fourth floor,blue,red,violet,154,35
3,P003,Haruto Tanaka,1996-10-27,lobby,green,blue,violet,158,29
4,P036,Kwame Mensah,1991-02-26,fourth floor,blue,yellow,orange,184,34


**TASK 2.3 (10 Points):** Now that we have created the `Age`; lets clean the `Eye_Color` column.

***Please read the directions carefully:***


- In order to fill in the missing eye colors, we want to find the **most common eye color** among those who have the **same hair and skin color**.
- For instance, if we want to find the eye color of someone who has yellow hair and violet skin:
 - we would first find everyone who has the same hair and skin color, in this case it would be yellow hair and violet skin
 - and then we find the most common eye color among them.
 - we then replace the missing eye color with the eye color that is the most common among the people with the same hair and skin color.
- Watch out for redundant data!

- ***NOTE: In the case that you find multiple modes, choose the color that appears first in alphabetical order***

 When you finish filling in the missing eye colors, please **display the `profiles_df`**  (only top 5 rows)

 **Outputs to be Graded**
 - Variable(s): `profiles_df`
 - Cell outputs

In [28]:
# Write your code here

#fucntion to fill the eye color of the missing person (person_row is a series)
def fill_missing_eye_color(person_row):
    person_hair = person_row["Hair_Color"]
    person_skin = person_row["Skin_Color"]
    eye_color_guess = series_mode[(person_hair, person_skin)] #using the person mode seires to look up the mode to guess 
    return eye_color_guess

non_redudant_df = profiles_df.drop_duplicates(subset="ID") #Drop all duplicates id

#group them by their haird clor and skin color and find which is the most common eye color 
series_mode = non_redudant_df.groupby(['Hair_Color', 'Skin_Color'])['Eye_Color'].agg(lambda x: x.mode().iloc[0])

#make a df with only ppl with missing eye colors for us to fill
na_eye_color_df = profiles_df[profiles_df['Eye_Color'].isna()]
series_na_eye_colors_filled = na_eye_color_df.apply(fill_missing_eye_color, axis = 1)

# Write each guessed eye color into the rows where Eye_Color is missing
profiles_df.loc[profiles_df['Eye_Color'].isna(), 'Eye_Color'] = series_na_eye_colors_filled

profiles_df.head()

,ID,Name,Birthday,Location_during_crime,Eye_Color,Skin_Color,Hair_Color,Height,Age
0,P299,Marco Rossi,1969-10-07,fourth floor,red,indigo,orange,191,56
1,P086,Aarav Patel,1979-10-03,second floor,indigo,indigo,green,198,46
2,P022,Li Wei,1990-08-10,fourth floor,blue,red,violet,154,35
3,P003,Haruto Tanaka,1996-10-27,lobby,green,blue,violet,158,29
4,P036,Kwame Mensah,1991-02-26,fourth floor,blue,yellow,orange,184,34



## Q3) Finding the features of the culprit (43 PTS Total)

Now that we have successfully cleaned all of our data, lets begin finding the features of the criminal. Take a moment to look at the `Witnesses` table. We don't necessarily want all the statements from every single witness.

**NOTE:** Take a look at the structure of each statement, notice how there are only a few variations of statements from the witnesses.

**TASK 3.1 (5 Points):** Find all witnesses that were near the crime scene
- First load in the `Witness.csv` file and set it to a variable called `witness_df`
- create a new dataframe called `important_witnesses` that contains the id and the witness statements of those who were near the scene of the crime
- display the new dataframe  (only top 5 rows) when you are finished
- **HINT:** look at the floor which the crime happened on.

**Outputs to be Graded:**
- Variable(s): `witness_df`, `important_witnesses`
- Cell outputs

In [57]:
# Write your code here

# Loading the witness csv
witness_df = pd.read_csv('Witness.csv')


#df of id and statment of ppl who wher enear the crime
second_floor_df = profiles_df[profiles_df['Location_during_crime'] == 'second floor']

#Getting all the rows from the witness df who id matches with ppl in the seocnd floor
important_witnesses =  witness_df[witness_df['ID'].isin(second_floor_df['ID'])]

#make it so only the id and statement column show
important_witnesses = important_witnesses[['ID', 'Statement']]

important_witnesses.head()

,ID,Statement
12,P150,I saw someone with orange hair acting suspicious.
16,P075,I saw someone with orange hair acting suspicious.
20,P058,I saw someone with green eyes acting suspicious.
21,P157,I saw someone who looked 152 cm tall acting su...
36,P139,I saw someone who looked 150 cm tall acting su...


**TASK 3.2 (10 Points):** Find every potential eye color that the culprit could have

 - Put all the potential eye colors of the culprit in a list called `potential_eye_colors` and **display** it when you're done.
 - `potential_eye_colors` should have **no duplicates.**

 **Outputs to be Graded:**
 - Variable(s): `potential_eye_colors`
 - Cell outputs

In [40]:
# Write your code here

#filter the important_witnesses to statements containing "eyes"
eye_df = important_witnesses[important_witnesses['Statement'].str.contains('eyes')]

#split the statements and store it in a series
split_statements = eye_df['Statement'].str.split()

#Get the fourht index whihc is the eye color
series_colors = split_statements.str[4]

#make it so there is no duplicates
series_colors = series_colors.unique()

#turn numpy array into a list
potential_eye_colors = series_colors.tolist()


potential_eye_colors 

['green', 'orange', 'blue', 'violet']

**TASK 3.3 (5 Points):** Find every potential hair color that the culprit could have

 - Put all the potential hair colors of the culprit in a list called `potential_hair_colors` and **display** it when you're done.

**Outputs to be Graded**
- Variables: `potential_hair_colors`
- Cell outputs

In [42]:
# Write your code here

#making a df with only hair statments
hair_df = important_witnesses[important_witnesses['Statement'].str.contains('hair')]

#split them into tokens
hair_series = hair_df['Statement'].str.split()

#get the hair index which is 4 
hair_series = hair_series.str[4]
#removes duplsicate with unique() which return numpy list
hair_series = hair_series.unique()

#turn it into a llist
potential_hair_colors = hair_series.tolist()

potential_hair_colors

['orange', 'green', 'blue', 'violet']

**TASK 3.4 (5 Points):** Find every potential skin color that the culprit could have

 - Put all the potential skin colors of the culprit in a list called `potential_skin_colors` and display it when you're done

**Outputs to be Graded**
- Variable(s): `potential_skin_colors`
- Cell outputs

In [46]:
# Write your code here
#make a df with only the skin
skin_df = important_witnesses[important_witnesses['Statement'].str.contains('skin')]
#split the statment into tokens
skin_serie = skin_df['Statement'].str.split()

#get the fourth index
skin_serie = skin_serie.str[4]
#making them unique
sking_serie = skin_serie.unique()


potential_skin_colors = sking_serie.tolist()

potential_skin_colors

['orange', 'red', 'green']

**TASK 3.5 (10 Points):** Find the potential height of the culprit

 - Put the smallest and largest potential heights of the culprit in the variables called `smallest_height` and `largest_height` and display them when you are done

 **Outputs to be Graded**
 - Variables: `smallest_height`, `largest_height`
 - Cell outputs

In [50]:
# Write your code here

#make a df with only statemnet contisng the height
height_df = important_witnesses[important_witnesses['Statement'].str.contains("cm")]
#make a series with split tokens
height_series = height_df['Statement'].str.split()
#get the fifth index of each serie 
height_series = height_series.str[5].astype(int)

#get min max height
smallest_height =  height_series.min()
largest_height =  height_series.max()

print(smallest_height)
print(largest_height)


150
170


**TASK 3.6 (8 Points):** Find the potential age of the culprit

 - Put the smallest and largest potential age of the culprit in the variables called `smallest_age` & `largest_age` and display them when you are done

 **Outputs to be Graded**
 - Variable(s): `smallest_age`, `largest_age`
 - Cell outputs

In [52]:
# Write your code here
#make a age df  with only the witness stamntes
age_df = important_witnesses[important_witnesses['Statement'].str.contains('years')]
#since we are using str split on a single colum retunr a series with split worrds
age_series = age_df['Statement'].str.split()
#here the age is index 7 or is it about 6-7
age_series = age_series.str[7].astype(int)

smallest_age = age_series.min()
largest_age =  age_series.max()


print(smallest_age)
print(largest_age)

21
35


## Q4) Who owns the Ticket? (30 PTS Total)

**TASK 4.1 (2 Points):** Loading final dfs


Load in the `Zichao_s_Magic_Show.csv` into a dataframe called `show_df` and the `Interrogation_Statements.csv` into a dataframe called `interrogation_df` respectively and display them (only top 5 rows) when you are done

**Outputs to be Graded**
- Variables: `show_df`, `interrogation_df`
- Cell outputs

In [62]:
# Write your code here
show_df = pd.read_csv('Zichao_s_Magic_Show.csv')
interrogation_df = pd.read_csv('Interrogation_Statements.csv')

print(show_df.head())
print(interrogation_df.head())

     ID  Ticket_Number
0  P132       50269975
1  P056       28120732
2  P250       88051289
3  P197       79545169
4  P012       83521845
                   Name                                          Statement
0  Kristopher Stoichkov  "Re%spe%cti%ng o%the%rs' be%lo%ngi%ngs i%s my ...
1         Naman Nagelia  "I@'d ne@ve@r je@o@pa@rdi@ze@ my re@pu@ta@ti@o...
2  Pranavh Vallabhaneni  "My pro^je^ct pre^se^nta^ti^o^n o^ccu^pi^e^d m...
3             Stacy Sun  "Thre$e$ co$nse$cu$ti$ve$ e$xa$ms dra$i$ne$d m...
4        Rina Kobayashi  "I% wa%s o%ff-ca%mpu%s a%tte%ndi%ng a% fa%mi%l...


We're **so close** to figuring out the *last* and *final* clue to locate the culprit, but it looks like they are trying to beat us to it! What a devious criminal!

The culprit must have **sabotaged** our data after **interrogation** because it looks so wonky with weird symbols.

However, HQ informed us that our good friend and exemplar cryptanalyst **Alan Turing** has something to say to help us fix the data. **He encoded his message to keep it safe from the culprit.**

**TASK 4.2 (10 Points):**

Find what **Alan Turing** has to say:
- Use the `interrogation_df` to locate Alan's Turing's message.
- Put the message in a variable called `encoded_message` as a string (**Exclude** any **quotation marks** within the statement)
 - **Print** the encoded message in the following format: {"Encoded Message:", encoded_message}
- Figure out what the message says by taking a look at the numbers:
 - A **single space** between numbers indicates a letter to form a word.
 - A **double space** between numbers indicates the start of a new word.
- Store his decoded message in a variable called `decoded_message`
- Print out **Alan Turing's** decoded message in the following format: {"Decoded Message:", decoded_message}

**Outputs to be Graded**
- Variable(s): `encoded_message`, `decoded_message`
- Cell outputs: The two **formatted** messages

In [90]:
# Write your code here

alaran_df = interrogation_df[interrogation_df['Name'] == 'Alan Turing']
encoded_message = alaran_df['Statement'].iloc[0]
encoded_message = encoded_message.replace('"', "")
#Split the whole setences into words
split_into_word = encoded_message.split("  ")
#split the space betwen the words
words = []
#make it into a single word
for word_with_single_space in split_into_word:
    words.append(word_with_single_space.split(" "))

#loop thrught each numebr and covert it into a letter
decoded_message = "" 
for word in words:
    for num in word:
        decoded_message += string.ascii_lowercase[int(num) - 1]
        
    
    decoded_message += " "
    
#strip any leading white spaces 
decoded_message = decoded_message.strip()
    
print({"Encoded Message:", encoded_message})
print({"Decoded Message:", decoded_message})

{'Encoded Message:', '20 8 5  16 1 20 20 5 18 14  9 19  20 15  3 12 5 1 14  21 16  20 8 5  6 9 12 5  2 25  18 5 13 15 22 9 14 7  1 6 20 5 18  5 1 3 8  22 15 23 5 12  20 8 5  19 25 13 2 15 12 19  16 5 18 3 5 14 20  4 15 12 12 1 18  3 1 18 18 15 20  1 20'}
{'the pattern is to clean up the file by removing after each vowel the symbols percent dollar carrot at', 'Decoded Message:'}


**TASK 4.3 (10 Points):** **Restore** the interrogation statements back to their original state after **following the decoded message from before**.
- After restoration, make sure your modifications are updated in the `interrogation_df` itself.
- **Display** the `interrogation_df` (only top 5 rows) when you are done.

**Outputs to be Graded:**
- Variable(s): `interrogation_df`
- Cell outputs

In [99]:
# Write your code here

interrogation_series = interrogation_df['Statement'].str.replace("[%@^$]", "", regex =True)

interrogation_df['Statement'] = interrogation_series


interrogation_df.head()

,Name,Statement
0,Kristopher Stoichkov,"""Respecting others' belongings is my principle..."
1,Naman Nagelia,"""I'd never jeopardize my reputation by stealin..."
2,Pranavh Vallabhaneni,"""My project presentation occupied me in the en..."
3,Stacy Sun,"""Three consecutive exams drained me that day. ..."
4,Rina Kobayashi,"""I was off-campus attending a family event. Se..."



**TASK 4.4 (8 Points):** Create a column called `Ticket_Color` for the `show_df` that displays all the colors of the ticket

- Remember that there was a show ticket left at the crime scene, take a second to look through `show_df`. Try and find a way to identify how you can find the ticket color from the ticket number.
- **Display** the `show_df`  (only top 5 rows) when you are done
- **HINT:** examine the `interrogation_df`

**Output to be Graded**
- Variable(s): `show_df`
- Cell outputs

In [103]:
# Write your code here
interrogation_df.head(50)


,Name,Statement
0,Kristopher Stoichkov,"""Respecting others' belongings is my principle..."
1,Naman Nagelia,"""I'd never jeopardize my reputation by stealin..."
2,Pranavh Vallabhaneni,"""My project presentation occupied me in the en..."
3,Stacy Sun,"""Three consecutive exams drained me that day. ..."
4,Rina Kobayashi,"""I was off-campus attending a family event. Se..."
5,Asmita Brahme,"""Midterms consumed my entire day. Theft requir..."
6,Vismay Igur,"""Campus access logs prove I wasn't present. I ..."
7,Hted Oo,"""Back-to-back exams left no room for distracti..."
8,Ankit Nakhawa,"""I own dozens of markers. Why risk my academic..."
9,Ivy Lu,"""My marker collection is extensive. Stealing D..."


## Q5) Finding the culprit (12 PTS Total)

A helpful function might be the "iloc" function. Have a look at: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.iloc.html

**TASK 5.1(10 Points):** Using all the **data compiled from before** find a person that matches all the features from the profile.

- Do not hardcode or you will lose credit after all your hardwork

- Put the name of the culprit in a variable called `culprit`
- **Print** out the name of the culprit (There is only one).

**Outputs to be Graded**
- Variable(s): `culprit`
- Cell outputs

In [ ]:
# Write your code here

**TASK 5.2 (2 Points):** Check if your culprit is correct!
- **Interrogate** your culprit to see if you got the correct person.
- **Print** out their entire statement.

**Outputs to be Graded:**
- Cell output (the entire statement)


In [ ]:
# Write your code here

# GREAT WORK DETECTIVE!!